syntax = "proto3";

package function;
option go_package = "./wf_pb";

message RequestBody {
    optional bytes data = 1;
    string workflow_id = 2;
    int32 depth = 3;
    int32 width = 4;
    optional string request_type = 5;
    optional string pv_path = 6;
}

message ResponseBody {
    optional string reply = 1;
    int32 code = 2;
    optional string pv_path = 3;
}

service gRPCFunction {
    rpc gRPCFunctionHandler(RequestBody) returns (ResponseBody);
}
